lib/historical_link_checker.rb

```diff
@@ -0,0 +1,77 @@
+class HistoricalLinkChecker
+  attr_reader :report
+
+  def initialize(uri, checks)
+    @uri = uri
+    @checks = sorted_checks(checks)
+    @report = LinkChecker::UriChecker::Report.new
+  end
+
+  def call
+    current_report = LinkChecker.new(uri).call
+
+    if needs_historical_data_check?(current_report)
+      check_history(current_report)
+    else
+      current_report
+    end
+  end
+
+private
+
+  attr_reader :uri, :checks
+
+  def check_history(current_report)
+    if all_checks_have_errors?(checks_since(7.days.ago))
+      current_report.add_problem(
+        recurred_for_more_than_one_week_error(current_report)
+      )
+    elsif all_checks_have_errors?(checks_since(4.days.ago))
+      current_report.add_problem(
+        recurred_for_more_than_three_days_error(current_report)
+      )
+    else
```

Comment:
Would you end up with two errors that are for the same thing e.g.

- This problem has been occurring for more than 3 days. Page Not Found
- Page Not Found 

?